### PR TITLE
Make existing python and python-dependent packages depend on python_abi

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by scopatz** [comment]
LGTM too

**Comment by mattip** [comment]
I am not familiar with the workflow here. Are there tests or docstrings that would help a new person understand when there are bugs in the new functions?

File: recipe/test_gen_patch_json.py [hunk 1]
```diff
@@ -27,3 +27,65 @@ def test_gen_patch_instructions():
         'c': {'addthis': 'yes'}}
 
     assert inst['remove'] == list(REMOVALS['osx-64'])
+
+
+def test_add_python_abi():
+    conditions = {
+        'python >=2.7,<2.8.0a0': 'cp27mu',
```
Comment (isuruf) [inline recipe/test_gen_patch_json.py]: @mattip, a package with the dependency given as `python >=2.7,<2.8.0a0` gets tagged with the abi tag `cp27mu`.

**Comment by ocefpaf** [comment]
Let's give this a go. Thanks @isuruf!

**Pull request merged**